### PR TITLE
Cleaned up a *lot* of places where errors are ignored or otherwise badly handled

### PR DESCRIPTION
**Pull request by GabrielRavier**
Note: You might want to see whether it might be better in some places to throw the error back up the call stack to end up being reported elsewhere, I might have possibly been overly careful in some places as I wanted to avoid the chances of false positives as much as possible (i.e. there's a bunch of places where I just log the error and do nothing else with it).

I also corrected a few task name typos in server while I was at it since they were right next to the errors, but do tell me if that should go into another PR

### CODE REVIEW

File: server/utils.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+//   Copyright 2017 MSolution.IO
```
Comment (cruffinoni) [inline server/utils.go]: Shouldn't you put 2021?

File: users/password.go [hunk 2]
```diff
@@ -78,18 +78,30 @@ func init() {
...
+	}
 	if err == nil {
-		now := time.Now()
-		expire := now.Add(time.Hour * time.Duration(-1*int64(nbHoursValidityForgottenToken)))
-		models.DeleteExpiredForgottenPassword(transaction, expire)
 		defer func() {
 			rec := recover()
-			if rec != nil {
-				transaction.Rollback()
+			if rec != nil || err != nil {
```
Comment (cruffinoni) [inline users/password.go]: `err` will be always null in this condition

**Review by cruffinoni — CHANGES_REQUESTED** [review]
I'm waiting for @simonmeyerrr or someone else to do a review before merging.

File: server/utils.go (hunk 1 above)
Comment (GabrielRavier) [reply server/utils.go]: Oops, that was a copy-paste error. Corrected it !

File: users/password.go (hunk 2 above)
Comment (GabrielRavier) [reply users/password.go]: It's in a deferred function so it can be non-nil after the call to `models.DeleteExpiredForgottenPassword`

**Review by simonmeyerrr — APPROVED** [review]

**Pull request merged**